src/tests/class_tests/openms/source/DIAPrescoring_test.cpp

```diff
@@ -271,7 +284,103 @@ START_SECTION ( test score function with shifted first transition )
 }
 END_SECTION
 
+START_SECTION ( test score function missing first transition due to different ion mobility )
+{
+  OpenSwath::LightTransition mock_tr1;
+  mock_tr1.product_mz = 500.;
+  mock_tr1.fragment_charge = 1;
+  mock_tr1.transition_name = "group1";
+  mock_tr1.library_intensity = 5.;
+
+  OpenSwath::LightTransition mock_tr2;
+  mock_tr2.product_mz = 600.;
+  mock_tr2.fragment_charge = 1;
+  mock_tr2.transition_name = "group2";
+  mock_tr2.library_intensity = 5.;
+
+
+  double PRECURSOR_ION_MOBILITY = 7;
+  double ION_MOBILITY_WIDTH = 2;
+
+  OpenSwath::SpectrumPtr sptr = (OpenSwath::SpectrumPtr)(new OpenSwath::Spectrum);
+
+  std::vector<OpenSwath::BinaryDataArrayPtr> binaryDataArrayPtrs;
+  OpenSwath::BinaryDataArrayPtr data1(new OpenSwath::BinaryDataArray);
+  OpenSwath::BinaryDataArrayPtr data2(new OpenSwath::BinaryDataArray);
+  OpenSwath::BinaryDataArrayPtr data3(new OpenSwath::BinaryDataArray);
+
+  static const double arr1[] = {
+
+    10, 20, 50, 100, 50, 20, 10, // peak at 499
+    3, 7, 15, 30, 15, 7, 3,      // peak at 500
+    1, 3, 9, 15, 9, 3, 1,        // peak at 501
+    3, 9, 3,                     // peak at 502
+
+
+    10, 20, 50, 100, 50, 20, 10, // peak at 600
+    3, 7, 15, 30, 15, 7, 3,      // peak at 601
+    1, 3, 9, 15, 9, 3, 1,        // peak at 602
+    3, 9, 3                      // peak at 603
+  };
+  std::vector<double> intensity (arr1, arr1 + sizeof(arr1) / sizeof(double) );
+  static const double arr2[] = {
+
+    498.97, 498.98, 498.99, 499.0, 499.01, 499.02, 499.03,
+    499.97, 499.98, 499.99, 500.0, 500.01, 500.02, 500.03,
+    500.97, 500.98, 500.99, 501.0, 501.01, 501.02, 501.03,
+    501.99, 502.0, 502.01,
+
+    599.97, 599.98, 599.99, 600.0, 600.01, 600.02, 600.03,
+    600.97, 600.98, 600.99, 601.0, 601.01, 601.02, 601.03,
+    601.97, 601.98, 601.99, 602.0, 602.01, 602.02, 602.03,
+    602.99, 603.0, 603.01
+  };
+  std::vector<double> mz (arr2, arr2 + sizeof(arr2) / sizeof(double) );
+
+  static const double arr3[] = {
+      1, 1, 3, 1, 1, 1, 1,      // peak at 499
+      2, 2, 3, 1, 2, 1, 2,      // peak at 500
+      1, 2, 1, 2, 3, 2, 2,      // peak at 501
+      2, 2, 2,                  // peak at 502
+
+      7, 6, 7, 6, 8, 6, 8,      // peak at 600
+      7, 7, 7, 8, 7, 7, 8,      // peak at 601
+      8, 6, 8, 8, 6, 6, 6,      // peak at 602
+      6, 8, 6                   // peak at 603
+  };
+  std::vector<double> im (arr3, arr3 + sizeof(arr3) / sizeof(double) );
+
+  data1->data = mz;
+  data2->data = intensity;
+  data3->data = im;
+  sptr->setMZArray( data1);
+  sptr->setIntensityArray( data2);
+  data3->description = "Ion Mobility";
```

Comment:
please dont

Reply:
Sorry @cbielow I'm confused what is wrong with this.

Reply:
Same problem as here:
https://github.com/OpenMS/OpenMS/pull/6479#discussion_r1071290079

;-)